superset/models/helpers.py

```diff
@@ -374,13 +374,20 @@ class QueryResult:  # pylint: disable=too-few-public-methods
     """Object returned by the query interface"""
 
     def __init__(  # pylint: disable=too-many-arguments
-        self, df, query, duration, status=QueryStatus.SUCCESS, error_message=None
+        self,
+        df,
+        query,
+        duration,
+        status=QueryStatus.SUCCESS,
+        error_message=None,
+        errors=None,
     ):
         self.df: pd.DataFrame = df
         self.query: str = query
         self.duration: int = duration
         self.status: str = status
         self.error_message: Optional[str] = error_message
+        self.errors: Optional[List[Dict[str, Any]]] = errors
```

Comment:
@etr2460 sorry this could be 

```python
self.errors: List[Dict[str, Any]] = errors or []
```

which simplifies the related content.

Reply:
I actually think it's better to keep this optional? Because at the end of the day we send `errors` down to the client, and i think it being null is better than `[]`

Reply:
@dpgaspar @villebro any thoughts on consistency here, i.e., should we use an empty list or `None` to represent no errors?

Reply:
I moved to empty list to make the backend logic a bit neater